packages/blob/src/list.ts

```diff
@@ -20,40 +20,59 @@ export interface ListBlobResult {
   hasMore: boolean;
 }
 
+export interface ListFoldedBlobResult extends ListBlobResult {
+  folders: string[];
+}
+
 interface ListBlobApiResponseBlob
   extends Omit<ListBlobResultBlob, 'uploadedAt'> {
   uploadedAt: string;
 }
 
 interface ListBlobApiResponse extends Omit<ListBlobResult, 'blobs'> {
   blobs: ListBlobApiResponseBlob[];
+  folders?: string[];
 }
 
-export interface ListCommandOptions extends BlobCommandOptions {
+export interface ListCommandOptions<
+  M extends 'expanded' | 'folded' | undefined = undefined,
+> extends BlobCommandOptions {
   /**
    * The maximum number of blobs to return.
    * @defaultvalue 1000
    */
   limit?: number;
   /**
-   * Filters the result to only include blobs located in a certain folder inside your store.
+   * Filters the result to only include blobs that start with this prefix.
+   * If used together with `mode: 'folded'`, make sure to include a trailing slash after the foldername.
    */
   prefix?: string;
   /**
    * The cursor to use for pagination. Can be obtained from the response of a previous `list` request.
    */
   cursor?: string;
+  /**
+   * Defines how the blobs are listed
+   * - `expanded` the blobs property contains all blobs.
+   * - `folded` the blobs property contains only the blobs at the root level of your store. Blobs that are located inside a folder get merged into a single entry in the folder response property.
+   * @defaultvalue 'expanded'
+   */
+  mode?: M;
 }
 
+type ListCommandResult<
+  M extends 'expanded' | 'folded' | undefined = undefined,
+> = M extends 'folded' ? ListFoldedBlobResult : ListBlobResult;
+
 /**
  * Fetches a paginated list of blob objects from your store.
  * Detailed documentation can be found here: https://vercel.com/docs/storage/vercel-blob/using-blob-sdk#list-blobs
  *
  * @param options - Additional options for the request.
  */
-export async function list(
-  options?: ListCommandOptions,
-): Promise<ListBlobResult> {
+export async function list<
+  M extends 'expanded' | 'folded' | undefined = undefined,
+>(options?: ListCommandOptions<M>): Promise<ListCommandResult<M>> {
```

Comment:
Can you screenshot the VSCode experience from your preview package, when using mode: folded or mode: expanded or no option?

Reply:
## No options, same as before
<img width="415" alt="Screenshot 2023-11-07 at 12 03 54" src="https://github.com/vercel/storage/assets/41523937/cc62210c-fe14-4d14-8291-de22f7413995">

## No options or `mode: expanded` doesn't expose folders
<img width="408" alt="Screenshot 2023-11-07 at 12 08 33" src="https://github.com/vercel/storage/assets/41523937/f579488f-7222-4bde-b332-df4c2c5f9343">

## All autocomplete fields
<img width="484" alt="Screenshot 2023-11-07 at 12 06 58" src="https://github.com/vercel/storage/assets/41523937/1fa9d4fd-66ff-4d06-9f10-03ce1c325971">

##  Autocomplete for mode union
<img width="523" alt="Screenshot 2023-11-07 at 12 07 50" src="https://github.com/vercel/storage/assets/41523937/037e6daf-b790-4155-ba81-b6cb6c2da5e3">

##  `mode: folded` exposes folders
<img width="399" alt="Screenshot 2023-11-07 at 12 08 44" src="https://github.com/vercel/storage/assets/41523937/b630f936-b477-4015-b932-8cd8af64199c">

##  And we also still have jsdoc for all the props
<img width="1136" alt="Screenshot 2023-11-07 at 12 12 05" src="https://github.com/vercel/storage/assets/41523937/f749a4b8-d516-46a1-b7d8-f76d732b0a17">
<img width="646" alt="Screenshot 2023-11-07 at 12 12 44" src="https://github.com/vercel/storage/assets/41523937/446ea2ab-c6a3-4f05-9409-e039798a561f">

Reply:
really nice work, thanks